### ISSUE TITLE
COPY inside plpgsql functions fails after first invocation?

### ISSUE THREAD

**Issue by lithp** [citusdata/citus#2127]
```
brian=# CREATE FUNCTION doit() RETURNS void AS $$
BEGIN
  COPY (SELECT 'hi') TO '/home/brian/Work/pg-10/some_file';
END;
$$ LANGUAGE plpgsql;
CREATE FUNCTION
brian=# SELECT doit();
 doit 
------
 
(1 row)

brian=# SELECT doit();
WARNING:  unrecognized node type: 2139062143
ERROR:  unexpected command tag "???"
CONTEXT:  SQL statement "COPY (SELECT 'hi') TO '/home/brian/Work/pg-10/some_file'"
PL/pgSQL function doit() line 3 at SQL statement
```

Even more tellingly:

```
brian=# CREATE FUNCTION doit() RETURNS void AS $$
BEGIN
  COPY (SELECT 'hi') TO '/home/brian/Work/pg-10/some_file';
END;
$$ LANGUAGE plpgsql;
CREATE FUNCTION
brian=# SELECT doit();
 doit 
------
 
(1 row)

brian=# SELECT doit();
 doit 
------
 
(1 row)

brian=# SELECT doit();
 doit 
------
 
(1 row)

brian=# CREATE EXTENSION citus;
CREATE EXTENSION
brian=# SELECT doit();
 doit 
------
 
(1 row)

brian=# SELECT doit();
WARNING:  unrecognized node type: 9
ERROR:  unexpected command tag "???"
CONTEXT:  SQL statement "COPY (SELECT 'hi') TO '/home/brian/Work/pg-10/some_file'"
PL/pgSQL function doit() line 3 at SQL statement
```

**Comment by lithp** [comment]
Copying the explanation over from #2128:

This happens because of some code in `multi_ProcessUtility` which [copies `parsetree`](https://github.com/citusdata/citus/blob/b33b28203071e39b7910c04c0493651b8e5003dc/src/backend/distributed/executor/multi_utility.c#L315):

```c
	if (IsA(parsetree, CopyStmt))
	{
		/* copy parse tree since we might scribble on it to fix the schema name */
		parsetree = copyObject(parsetree);

		parsetree = ProcessCopyStmt((CopyStmt *) parsetree, completionTag,
									&commandMustRunAsOwner);

                ...
	}
```

and then [assigns the copy to pstmt->utilityStmt](https://github.com/citusdata/citus/blob/b33b28203071e39b7910c04c0493651b8e5003dc/src/backend/distributed/executor/multi_utility.c#L488):

```c
	pstmt->utilityStmt = parsetree;
	standard_ProcessUtility(pstmt, queryString, context,
							params, queryEnv, dest, completionTag);
```

(this code was added in https://github.com/citusdata/citus/commit/853f07dd33fbe8c243337c663025714022c61e82, which fixed `COPY` in `plpgsql` for distributed tables, but broke it for local tables.)

The problem is that, when we're inside plpgsql the plan gets cached and reused between statements. It belongs to a memory context which is [a child of CacheMemoryContext](https://github.com/postgres/postgres/blob/1222db999dc8ad055e0320dd6704d814acca3b51/src/backend/utils/cache/plancache.c?utf8=%E2%9C%93#L1176). When we call `copyObject`, we create a copy which lives in the local executor context, [which is reset](https://github.com/postgres/postgres/blob/1222db999dc8ad055e0320dd6704d814acca3b51/src/backend/executor/spi.c?utf8=%E2%9C%93#L2495) at the end of the statement.

**Comment by lithp** [comment]
Fixed with #2128
